add number formatting and pricing data

wesbos opened the pull request:
Add number formatting - commas a nice to read large values.

Also added pricing data for some common models. 

<img width="929" alt="Screenshot 2023-05-26 at 1 37 25 PM" src="https://github.com/niieani/gpt-tokenizer-demo/assets/176013/574577ed-5f95-4fd1-bc32-4ddd284dc50c">

wesbos committed:
add number formatting and pricing data

niieani commented on src/Stat.tsx:
```diff
@@ -0,0 +1,62 @@
+
+const pricePerToken = {
+  'gpt-3.5-turbo': 0.006 / 1000,
+  'gpt4': {
```
nit pick: missing dash between gpt and 4. Ideally, let's use official model names.

```suggestion
  'gpt-4': {
```

niieani commented on src/Stat.tsx:
```diff
@@ -0,0 +1,62 @@
+    prompt: 0.03 / 1000,
+    completion: 0.06 / 1000,
+  },
+  'gpt4-32k': {
```
```suggestion
  'gpt-4-32k': {
```

niieani reviewed (changes requested):
Hey @wesbos! Thanks for this, looks like very useful additions.
Overall looks great, small remarks:
- The prices for gpt-4* and gpt-3.5-turbo should only be displayed when `cl100k_base` is the selected encoding. Other encodings are used by older models (e.g. `text-davinci-003`) which have different prices, so the cost would be inaccurate after the switch. I'd suggest to either add prices for other models, or just hide them when prices are unavailable. You can import the `modelToEncodingMap` const from [`gpt-tokenizer/mapping`](https://github.com/niieani/gpt-tokenizer/blob/774cf367d8b6737376437b9bd9866bc143e8b33a/src/mapping.ts#L14)
- As per our discussion on Twitter, `claude`* models use a different tokenization strategy. I'm happy to accept a PR to `gpt-tokenizer` repo to include the tokenization strategy for anthropic's models (could be ported from their python tokenizer), but for now, these would be inaccurate, so I would suggest removing them.

wesbos committed:
Update src/Stat.tsx

Co-authored-by: Bazyli Brzóska <bazyli.brzoska@gmail.com>

wesbos committed:
Update src/Stat.tsx

Co-authored-by: Bazyli Brzóska <bazyli.brzoska@gmail.com>

wesbos committed:
conditionall show pricing based on encoding

wesbos commented:
thanks for the feedback - Ive implemented the requested changes, also added a description to each, and added the embeddings engine. Might be a bigger project to eventually show all pricing data - but this is handy enough for now!

niieani reviewed (approved):
Thanks!
